lightningd/peer_control.c

```diff
@@ -2734,8 +2734,15 @@ const char *peer_state_name(enum peer_state state)
 
 static void activate_peer(struct peer *peer)
 {
+	u8 *msg;
+
 	assert(!peer->owner);
 
+	/* Pass gossipd any addrhints we currently have */
+	msg = towire_gossipctl_peer_addrhint(peer, &peer->id, &peer->addr);
+	subd_send_msg(peer->ld->gossip, msg);
+	tal_free(msg);
```

Comment:
This works, but subd_send_msg() supports take, so you can simply tell it `subd_send_msg(peer->ld->gossip, take(msg));` and it will take ownership.  You can see that we did that below, for example.